vllm/transformers_utils/tokenizer.py

```diff
@@ -145,6 +216,94 @@ async def get_lora_tokenizer_async(
             return self.lora_tokenizers.get(lora_request.lora_int_id)
 
 
+if ray:
+    RayTokenizerGroup = ray.remote(TokenizerGroup)
+
+    class RayTokenizerGroupPool(BaseTokenizerGroup):
+        """A pool of TokenizerGroups for async tokenization."""
+
+        def __init__(  # pylint: disable=super-init-not-called
+                self, tokenizer_id: str, enable_lora: bool, max_num_seqs: int,
+                max_input_length: Optional[int], num_actors: int,
+                ray_actor_options: dict, **tokenizer_config):
+            self.tokenizer = TokenizerGroup(tokenizer_id, enable_lora,
+                                            max_num_seqs, max_input_length,
+                                            **tokenizer_config)
+            self.max_input_length = max_input_length
+
+            # Carry over the env vars to the actors.
+            # This is necessary for API keys and such.
+            ray_actor_options.setdefault("runtime_env", {})
+            env_vars = os.environ.copy()
+            ray_actor_options["runtime_env"].setdefault("env_vars", {})
+            env_vars.update(ray_actor_options["runtime_env"]["env_vars"])
+            ray_actor_options["runtime_env"]["env_vars"] = env_vars
```

Comment:
This is the kind of logic which should be unit tested (can be placed in a helper function to reduce test mock complexity)